utils/python/CIME/compare_test_results.py

```diff
@@ -60,39 +67,74 @@ def compare_test_results(baseline_name, baseline_root, test_root, compiler, test
                 caseroot = test_dir,
                 sfile = logfile_name)
 
-            run_result = ts.get_status(RUN_PHASE)
-            compare_comment = ""
-            detailed_comments = ""
-            if (run_result is None):
-                compare_result = "SKIP"
-                compare_comment = "Test did not make it to run phase"
-                do_compare = False
-            elif (run_result != TEST_PASS_STATUS):
-                compare_result = "SKIP"
-                compare_comment = "Run phase did not pass"
-                do_compare = False
+            if (not hist_only):
```

Comment:
No need to rework this code now, but a general comment for the sake of future code: I find this `if (not XXX_only)` logic (which appears in many places) to be confusing. It would be especially confusing and error-prone if there were more than two pieces. I'd find this sort of code easier to read, write and maintain if this were flipped around, so this line (for example) became `if (do_namelist_compare)`. Again, no need to rework this now, but I'd like us to consider that for the future.